sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -13,10 +13,32 @@
 
 #include "pi_level_zero.hpp"
 #include "ur_bindings.hpp"
+#ifdef _WIN32
+#include <Windows.h>
+#include <delayimp.h>
+#endif
 
 // Defined in tracing.cpp
 void enableZeTracing();
 void disableZeTracing();
+#ifdef _WIN32
+FARPROC WINAPI delayHook(unsigned dliNotify, PDelayLoadInfo pdli) {
+  switch (dliNotify) {
+  case dliStartProcessing:
+    break;
+  case dliNotePreLoadLibrary: {
+    if (strcmp(pdli->szDll, "ze_loader.dll") == 0) {
+      return (FARPROC)LoadLibraryExA("ze_loader.dll", nullptr,
```

Comment:
are you doing the same to ur_loader?

Reply:
No I am not sure/familiar if ur_loader also has load-time linking but this change is specific for ze_loader.

Reply:
Same as ze_loader: https://github.com/intel/llvm/blob/1060b20df94a32d3f23f85a639c763048a9aa909/sycl/plugins/unified_runtime/CMakeLists.txt#L149

Reply:
I think that could be changed when we mirror these changes for UR.